### ISSUE TITLE
you can delete chapter with pages 

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#3514]
You can delete chapter with pages.
![image](https://user-images.githubusercontent.com/5429548/47361641-c0261e80-d6d2-11e8-88ed-800170ed538d.png)

![image](https://user-images.githubusercontent.com/5429548/47361814-2e6ae100-d6d3-11e8-989d-441b61aa5764.png)

Problem is that chapter ID is still in  page data so you can't relay on this field to test if page has chapter.
![image](https://user-images.githubusercontent.com/5429548/47361682-d502b200-d6d2-11e8-95f2-3697270367b5.png)

**Comment by Jimmi08** [comment]
It used here:
Maybe reason why sometimes  is page url not correct.
```
	function sc_cpageurl()
	{
		$route = ($this->var['page_chapter'] == 0) ? 'page/view/other' : 'page/view';
		$urldata = $this->var;

		if($this->var['page_chapter'] && $this->chapterData[$this->var['page_chapter']])
		{
			$chapter = $this->chapterData[$this->var['page_chapter']]; 
			$urldata = array_merge($this->var, $chapter);
			$urldata['book_sef'] = $this->chapterData[$chapter['chapter_parent']]['chapter_sef'];
		}

		return e107::getUrl()->create($route, $urldata, array('allow' => 'page_sef,page_id,chapter_sef,book_sef'));
	}
```

**Comment by Jimmi08** [comment]
@Moc this is bug.  You can't allow to delete something if there is related table without checking. Not mention if you relay further in code that there is correct value.

**Comment by rica-carv** [comment]
Indeed.At least a warning that the page you're about to delete has related chapters....

**Comment by Jimmi08** [comment]
@rica-carv Page is last in tree, problem is deleted chapter with pages.

**Comment by rica-carv** [comment]
@Jimmi08 Humm, let's see if i understand. Chapter is a child of page, right? So why can't i delete a child of a parent page? It should always be allowed, but also, a warning could be in hand, but just to be sure if delete is intencional or not, nothing else....

**Comment by Jimmi08** [comment]
no, chapter is parent of page. Book is parent of chapter. Otherwise you are right.

**Comment by Jimmi08** [comment]
I would add this as a preference.  Or any visible info that chapter has pages. Because main problem is not deleting itself, but the case when you think that there are no pages and in fact, there are some. The nice way is used in the metatag plugin (with modal confirmation).
